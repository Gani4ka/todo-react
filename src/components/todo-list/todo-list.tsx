import React from 'react';
import TodoListItem from '../todo-list-item';
import './todo-list.css';

export interface TodoListProps {
  todos: Array<{
    id: string;
    important: boolean;
    done: boolean;
    label: string;
  }>;
  onDeleted: Function;
  onToggleImportant: Function;
  onToggleDone: Function;
}
export const TodoList = ({
  todos,
  onDeleted,
  onToggleImportant,
  onToggleDone
}: TodoListProps): JSX.Element => {
  const elements = todos.map(item => {
    const { id, ...itemProps } = item;
    return (
      <li key={id} id={id} className='list-group-item'>
        <TodoListItem
          {...itemProps}
          onDeleted={(): React.MouseEventHandler => onDeleted(id)}
          onToggleImportant={(): React.MouseEventHandler =>
            onToggleImportant(id)
          }
          onToggleDone={(): React.MouseEventHandler => onToggleDone(id)}
        />
      </li>
    );
  });

  return <ul className='list-group todo-list'>{elements}</ul>;
};
