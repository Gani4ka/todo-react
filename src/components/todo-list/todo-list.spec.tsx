import React from 'react';
import { shallow } from 'enzyme';
import { TodoList, TodoListProps } from './todo-list';

describe('<TodoList />', (): void => {
  const props: TodoListProps = {
    todos: [
      {
        label: '1',
        important: false,
        done: false,
        id: 1
      },
      {
        label: '2',
        important: false,
        done: false,
        id: 2
      }
    ],
    onDeleted: jest.fn(),
    onToggleImportant: jest.fn(),
    onToggleDone: jest.fn()
  };

  it('should render correctly', (): void => {
    const component = shallow(<TodoList {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should render no items', (): void => {
    props.todos = [];
    const component = shallow(<TodoList {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });
});
