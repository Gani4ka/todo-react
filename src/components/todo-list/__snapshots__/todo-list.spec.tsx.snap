// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should render correctly 1`] = `
"<ul className=\\"list-group todo-list\\">
  <li id=\\"1\\" className=\\"list-group-item\\">
    <TodoListItem label=\\"1\\" important={false} done={false} onDeleted={[Function: onDeleted]} onToggleImportant={[Function: onToggleImportant]} onToggleDone={[Function: onToggleDone]} />
  </li>
  <li id=\\"2\\" className=\\"list-group-item\\">
    <TodoListItem label=\\"2\\" important={false} done={false} onDeleted={[Function: onDeleted]} onToggleImportant={[Function: onToggleImportant]} onToggleDone={[Function: onToggleDone]} />
  </li>
</ul>"
`;

exports[`<TodoList /> should render no items 1`] = `"<ul className=\\"list-group todo-list\\" />"`;
